### PR TITLE
[B5] Migrate case_search views

### PR DESCRIPTION
**Pull request by jingcheng16**
## Technical Summary

This PR migrates two views of `case_search`: `CaseSearchView` and `ProfileCaseSearchView`
I did migrate view and un-split view files in one PR, considering this is a small app. 
Please review by commit.

Form for Search Parameters
Before:
image
After:
image

Results Tab
Before:
image
After:
image

Query Tab
Before:
image
After:
image

Profile Tab
Before:
image
After:
image

Profile Case Search Page
Before:
image
After:
image

## Feature Flag

(required) Turn on the "[Enable extension sync](https://www.commcarehq.org/hq/flags/edit/extension_sync/?__hstc=240960668.2195ac63c0056c39905e6882865ae8e3.1597171287126.1616447050240.1616526330286.228&__hssc=240960668.8.1616526330286&__hsfp=1126783430)" feature flag

(required) Turn on the "[Enable synchronous mobile searching and case claiming](https://www.commcarehq.org/hq/flags/edit/search_claim/)" feature flag

### Safety story

I tested locally and on staging.  

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json
Comment (jingcheng16) [inline corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json:13]: This is weird, I think this line should be `case_search.html`... But this is generated by migration tool... so I don't know if this is a mistake by migration tool or it should be `.html`?

**Review by esoergel — COMMENTED** [review]
I don't see the changes from https://github.com/dimagi/commcare-hq/pull/34400 reflected in the `profile_case_search.html` diff - was this PR perhaps based off an earlier `master`?

File: corehq/apps/case_search/templates/case_search/bootstrap3/profile_case_search.html
Comment (esoergel) [inline corehq/apps/case_search/templates/case_search/bootstrap3/profile_case_search.html:65]: Ah, I see the queries section is present here, but not in the `profile_case_search.html` that remains

File: corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html
Comment (MartinRiese) [inline corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html:30]: I am curious if the nesting here is necessary. When I look at the [documentation](https://getbootstrap.com/docs/5.0/forms/layout/#horizontal-form-label-sizing) is looks like `col-auto` could just go on the label and you could remove this extra div level

File: corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html
Comment (MartinRiese) [inline corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html:93]: do we have a guide for adding margin like this? Or did you just add it make it look better?

File: corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html
Comment (MartinRiese) [inline corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html:99]: same here. do we need the new div or can col go on the textarea?

File: corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json
Comment (MartinRiese) [reply corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json:13]: I think this file is just to keep track of what work is left. And it does not seem to track all the other apps the Jenny did so far. So I would not worry to much about it @biyeun can you confirm this?

**Review by MartinRiese — APPROVED** [review]

**Comment by MartinRiese** [comment]
@jingcheng16 meant to comment not approve.

File: corehq/apps/case_search/templates/case_search/case_search.html
Comment (orangejenny) [inline corehq/apps/case_search/templates/case_search/case_search.html:99]: We usually target a medium screen size, so this would be `col-md-9`, so I'd do that for consistency's sake. (I searched through the bootstrap docs for a few min before finding that this targets extra small screens.)

**Review by orangejenny — COMMENTED** [review]
Looks good. I left some minor comments.

File: corehq/apps/case_search/templates/case_search/case_search.html
Comment (esoergel) [reply corehq/apps/case_search/templates/case_search/case_search.html:98]: :+1: `mb-1` would be nice - it bothered me a bit when I was fiddling with it, but I didn't want to add inline styles and wasn't sure of a decent alternative

**Review by esoergel — APPROVED** [review]

File: corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html
Comment (jingcheng16) [reply corehq/apps/case_search/templates/case_search/bootstrap5/case_search.html:93]: Hi Martin, I just add it to make it look better. I think the `panel` used to have margin or padding, but `card` doesn't have anything. Without adding it, the three cards will have no spacing between them at all.

File: corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json
Comment (jingcheng16) [reply corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5_completed.json:13]: I think Biyeun just added this feature recently, and other apps Jenny migrated is before this PR getting merged. Correct me if I'm wrong. @biyeun

File: corehq/apps/case_search/templates/case_search/bootstrap3/profile_case_search.html
Comment (jingcheng16) [reply corehq/apps/case_search/templates/case_search/bootstrap3/profile_case_search.html:65]: Hi Ethan, good finding. This panel is removed in split files stage... I'm still looking into why it happened. 😭 This is the commit db1b8b64b226f after I first merge master into my branch, and from the diff we can see your change is marked as diff...

**Comment by esoergel** [comment]
Just noticed this was still open.  @gherceg FYI this PR will probably conflict with profiler changes you're looking at making.
@jingcheng16 anything you're blocked on to merge this in?

**Comment by jingcheng16** [comment]
@esoergel I will close this PR and have a new PR for migrating views.

**Pull request closed without merging**
